fitbenchmarking/controllers/dfo_controller.py

```diff
@@ -32,18 +46,7 @@ def __init__(self, cost_func):
         self._soln = None
         self._popt = None
         self._pinit = None
-        self.algorithm_check = {
-            'all': ['dfogn', 'dfols'],
-            'ls': ['dfogn', 'dfols'],
-            'deriv_free': ['dfogn', 'dfols'],
-            'general': [],
-            'simplex': [],
-            'trust_region': ['dfols', 'dfogn'],
-            'levenberg-marquardt': [],
-            'gauss_newton': ['dfogn'],
-            'bfgs': [],
-            'conjugate_gradient': [],
-            'steepest_descent': []}
+        self.algorithm_check = self.ALGORITHM_CHECK
```

Comment:
This seems unnecessary, is there a reason not to just rename `ALGORITHM_CHECK` to `algorithm_check` and remove this line?

Reply:
I've tried making this change, but then for all minimizers I get this error message :
`Minimizer cannot be run with Controller with current "algorithm_type" option set.
Details: For the selected software, there are no minimizers with the algorithm type(s) selected in the options file`

So I think that `algorithm_check` in the base controller isn't being overridden?

Reply:
Was the default in base_controller still in the __init__? That would override it.

Reply:
Ah yes, this was the problem!